core/blockchain.go

```diff
@@ -1779,3 +1786,47 @@ func (bc *BlockChain) SubscribeLogsEvent(ch chan<- []*types.Log) event.Subscript
 func (bc *BlockChain) SubscribeBlockProcessingEvent(ch chan<- bool) event.Subscription {
 	return bc.scope.Track(bc.blockProcFeed.Subscribe(ch))
 }
+
+// GetAncestorHash return the hash of ancestor at the given number
+func (bc *BlockChain) GetAncestorHash(ref *types.Header, target uint64) common.Hash {
+	number := ref.Number.Uint64() - 1
+	hash := ref.ParentHash
```

Comment:
Nitpick, either use `var ( ... )` to make these align, or better yet use a simultaneous assignment like below (`number, hash := ref.Number.Uint64() - 1, ref.ParentHash`).